motulator/grid/utils/_plots.py

```diff
@@ -0,0 +1,318 @@
+
+# %%
+from types import SimpleNamespace
+import numpy as np
+import matplotlib.pyplot as plt
+from cycler import cycler
+
+from motulator.common.utils import (complex2abc)
+
+# Plotting parameters
+plt.rcParams['axes.prop_cycle'] = cycler(color='brgcmyk')
+plt.rcParams['lines.linewidth'] = 1.
+plt.rcParams['axes.grid'] = True
+plt.rcParams.update({"text.usetex": False})
+
+
+# %%
+# pylint: disable=too-many-branches
+def plot_grid(
+        sim, base=None, plot_pcc_voltage=False, plot_w=False, t_span=None):
+    """
+    Plot example figures of grid converter simulations.
+
+    Plots figures in per-unit values, if the base values are given. Otherwise
+    SI units are used.
+
+    Parameters
+    ----------
+    sim : Simulation
+        Should contain the simulated data.
+    base : BaseValues, optional
+        Base values for scaling the waveforms.
+    plot_pcc_voltage : bool, optional
+        'True' if the user wants to plot the 3-phase waveform at the PCC. This
+        is an optional feature and the grid voltage is plotted by default.
+    plot_w : bool, optional
+        'True' if the user wants to plot the grid frequency instead of the
+        phase angles (by default).
+    t_span : 2-tuple, optional
+        Time span. The default is (0, sim.ctrl.ref.t[-1]).
+
+    """
+    FS = 16  # Font size of the plots axis
+    FL = 16  # Font size of the legends only
+    LW = 3  # Line width in plots
+
+    mdl = sim.mdl  # Continuous-time data
+    ctrl = sim.ctrl.data  # Discrete-time data
+    ctrl.t = ctrl.ref.t  # Discrete time
+
+    # Check if the time span was given
+    if t_span is None:
+        t_span = (0, ctrl.t[-1])  # Time span
+
+    # Check if the base values were given
+    if base is None:
+        pu_vals = False
+        # Scaling with unity base values except for power use kW
+        base = SimpleNamespace(w=1, u=1, i=1, p=1000)
+    else:
+        pu_vals = True
+
+    # 3-phase quantities
+    #i_c_abc = complex2abc(mdl.converter.data.i_cs).T
+    i_g_abc = complex2abc(mdl.grid_filter.data.i_gs).T
+    e_g_abc = complex2abc(mdl.grid_model.data.e_gs).T
+    u_g_abc = complex2abc(mdl.grid_filter.data.u_gs).T
+
+    # Calculation of active and reactive powers
+    #p_g = 1.5*np.asarray(np.real(ctrl.fbk.u_g*np.conj(ctrl.fbk.i_c)))
+    #q_g = 1.5*np.asarray(np.imag(ctrl.fbk.u_g*np.conj(ctrl.fbk.i_c)))
+    p_g = 1.5*np.asarray(
+        np.real(mdl.grid_filter.data.u_gs*np.conj(mdl.grid_filter.data.i_gs)))
+    q_g = 1.5*np.asarray(
+        np.imag(mdl.grid_filter.data.u_gs*np.conj(mdl.grid_filter.data.i_gs)))
+    p_g_ref = np.asarray(ctrl.ref.p_g)
+    q_g_ref = np.asarray(ctrl.ref.q_g)
+
+    # %%
+    fig, (ax1, ax2, ax3) = plt.subplots(3, 1, figsize=(10, 7))
+
+    if not sim.ctrl.dc_bus_volt_ctrl:
+        if not plot_pcc_voltage:
+            # Subplot 1: Grid voltage
+            ax1.plot(mdl.grid_model.data.t, e_g_abc/base.u, linewidth=LW)
+            ax1.legend([r'$e_g^a$', r'$e_g^b$', r'$e_g^c$'],
```

Comment:
This is a matter of taste, but I would avoid using latex in the plots. Some potential users may not be familiar with latex and it may also cause some issues in certain systems. In other words, the default plots could be kept very simple. The style could be the same as in the drive plots, i.e., maybe the linewidth etc. options could also be avoided (since in laptops with poor screen resolutions the matplotlib's defaults might be better than ours). 

Later, we could have more advanced latex-based plots available in a feature branch (or something similar)? They can be useful for plotting the figs for papers, if the font and figure sizes are preset for double-column paper format.